src/scipp/integrate/__init__.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+"""Sub-package for integration."""
+
+from ..core import array, DataArray
+from ..compat.wrapping import wrap1d
+
+from typing import Callable
+
+
+def _integrate(func: Callable, da: DataArray, dim: str, **kwargs) -> DataArray:
+    if 'dx' in kwargs:
+        raise ValueError("Invalid argument 'dx': Spacing of integration points is "
+                         f"given by the '{dim}' coord.")
+    integral = func(x=da.coords[dim].values, y=da.values, **kwargs)
+    return DataArray(data=array(
+        dims=da[dim, 0].dims, values=integral, unit=da.unit * da.coords[dim].unit))
+
+
+@wrap1d()
+def trapezoid(da: DataArray, dim: str, **kwargs) -> DataArray:
+    """Integrate data array along the given dimension with the composite trapezoidal rule.
+
+    This is a wrapper around :py:class:`scipy.integrate.trapezoid`.
```

Comment:
Would it be better to not mention scipy and instead explain all details here? (including not using kwargs) This would allow us to implement it differently in the future. And it would make it easier for users to understand the functions without looking at external docs.

Reply:
I think there are big pros and cons on both sides. What do you think @nvaytet?

Is it too early too decide?